test/unit/math/rev/mat/fun/adj_jac_apply_test.cpp

```diff
@@ -1,12 +1,1198 @@
 #include <stan/math/rev/core.hpp>
 #include <test/unit/math/rev/mat/util.hpp>
 #include <gtest/gtest.h>
+#include <algorithm>
 #include <sstream>
+#include <tuple>
+#include <vector>
 
+/**
+ * Test that all types that are documented to be supported can actually be
+ * included in functor. Check initialization of is_vars_ and offsets_ are to
+ * specification.
+ */
+struct WeirdArgumentListFunctor1 {
```

Comment:
[optional]
This doesn't need to be rewritten, but going forward, more smaller tests are a lot easier to deal with and a lot easier to use to diagnose problems.

Reply:
Yeah, I agree. I made it big and bulky cause I thought that'd be more likely to cause a breakdown in the components though. But I guess that's a double-edged technique.